### DISCUSSION TITLE
How connection is managed between prisma's connection_limit and pgbouncer connection limit

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#13250] [Q&A]
Currently we have enabled pgbouncer (which helps with extended database connections), and increased connection_limit to 100, still I am getting connection timeout and I checked active connections raise to 65 only.

So I wanted to know how prisma is managing connections help from pgbouncer?

**Comment by austincrim** [comment]
Hey @nilkothari 👋 ,

Per our engineering teams, `connection_limit` configures Prisma's own internal pool and `pgbouncer=true` makes Prisma send it's queries in a way that is compatible with pgBouncer, so the two are mostly unrelated.

Where/how are you seeing the 65 active connections?

How many connections is pgBouncer configure to handle?

Do you have enough load to exhaust the Prisma `connection_limit`?

**Reply by ghost** [reply]
okay, so does that mean I can increase number of connection_limit which can be handled by pgBouncer?

we are using azure for deployment, and on azure monitor active connections were 65 and at code level connection_limit was 100 and still it was throwing connection timeout error.

So yes, load was there to exhaust the Prisma conneciton_limit.

**Reply by austincrim** [reply]
Do you know if your queries are taking a long time to execute?

You could also try [increasing the `pool_timeout`](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#increasing-the-pool-timeout) query parameter, the default is 10 seconds.
